hibernate-core/src/main/java/org/hibernate/service/internal/AbstractServiceRegistryImpl.java

```diff
@@ -184,6 +185,14 @@ private void registerAlternate(Class<?> alternate, Class<?> target) {
 
 	@Override
 	public <R extends Service> R getService(Class<R> serviceRole) {
+		//Fast-path for ClassLoaderService as it's extremely hot during bootstrap
+		//(and after bootstrap service loading performance is less interesting as it's
+		//ideally being cached by long term consumers)
+		if ( ClassLoaderService.class.equals( serviceRole ) ) {
```

Comment:
I think you can even use `==` instead of `equals` here

Reply:
Good point, thanks!

Reply:
https://github.com/hibernate/hibernate-orm/pull/5577